sycl/include/CL/sycl/handler.hpp

```diff
@@ -830,7 +806,17 @@ class __SYCL_EXPORT handler {
       // will yield a rounded-up value for the total range.
       size_t NewValX =
           ((NumWorkItems[0] + GoodFactorX - 1) / GoodFactorX) * GoodFactorX;
-      if (getenv("SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE") != nullptr)
+
+      // Check whether rounding trace has been requested.
+      // Call getenv only once.
+      static bool RoundingTraceChecked = false;
+      static bool RoundingTrace = false;
+      if (!RoundingTraceChecked) {
+        RoundingTrace =
+            getenv("SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE") != nullptr;
+        RoundingTraceChecked = true;
+      }
+      if (RoundingTrace)
```

Comment:
This is definitely a subject of thread-**un**safety pattern.
It's more convenient to use `SYCLConfig` class.
If needed its thread-safety can be handled in distinct patch.

Reply:
Yes, I'll switch to using the SYCLConfig infrastructure, preserve all existing env vars, and see if that is efficient enough.

Reply:
The SYCLConfig features are available from the SYCL library. They don't seem to be directly available in the headers, e.g. handler.hpp. Will need to go from header into library to use SYCLConfig.

Reply:
No change has been made in environment variables supported.
Use of getenv has been minimized, which reduces overhead when short-running kernels are executed on Windows. (Execution time of getenv appears to be greater than two orders of magnitude higher on Windows than Linux).
Thread safety is not an issue since values are only written by the processing of env var settings, and any  concurrent writes will write the same values.